### PR TITLE
Second try at duplicate IDs: supported except in specific family methods

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: src/main/scala/is/hail/methods/PCRelate.scala [hunk 1]
```diff
@@ -170,6 +170,8 @@ class PCRelate(maf: Double, blockSize: Int) extends Serializable {
     gt != 0.0 && gt != 1.0 && gt != 2.0
 
   def apply(vds: VariantDataset, pcs: DenseMatrix): Result[M] = {
+    vds.requireUniqueSamples("pc_relate")
```
Comment (tpoterba) [inline src/main/scala/is/hail/methods/PCRelate.scala]: I don't think so - PCA just adds a sample annotation

Comment (tpoterba) [inline src/main/scala/is/hail/methods/PCRelate.scala]: pc-relate / IBD generate matrices with string keys, and I think that breaks (at least better safe than sorry)

File: python/hail/dataset.py [hunk 2]
```diff
@@ -109,27 +109,6 @@ def sample_ids(self):
             self._sample_ids = jiterable_to_list(self._jvds.sampleIds())
         return self._sample_ids
 
-    @property
```
Comment (tpoterba) [inline python/hail/dataset.py]: you can get out the sample annotations as a key table!

File: src/main/scala/is/hail/utils/package.scala [hunk 3]
```diff
@@ -495,9 +495,31 @@ package object utils extends Logging
...
+      uniques += smod
+      b += smod
+    }
+
+    b.result() -> mapping.result()
+  }
 }
 
-// FIXME: probably resolved in 3.6 https://github.com/json4s/json4s/commit/fc96a92e1aa3e9e3f97e2e91f94907fdfff6010d
+// FIXME: probably resolved in 3.6 `https://github.com/json4s/json4s/commit/fc96a92e1aa3e9e3f97e2e91f94907fdfff6010d
```
Comment (tpoterba) [inline src/main/scala/is/hail/utils/package.scala]: it was not

File: python/hail/dataset.py (hunk 2 above)
Comment (jigold) [inline python/hail/dataset.py]: Ok. It looks like variant_annotations is already gone from VDS.

**Pull request merged**